### ISSUE TITLE
🐛 Bug Report — Runtime APIs: R2Object.writeHttpMetadata() for cacheExpiry throws

### ISSUE THREAD

**Issue by leader22** [cloudflare/workerd#851]
See this minimal repro.

```js
export default {
  async fetch(req, env) {
    // ...

    const r2o = await env.TEST_R2.put("X", "Y", {
      httpMetadata: {
        cacheExpiry: new Date(),
      },
    });

    const h = new Headers();
    r2o.writeHttpMetadata(h); // 💥

   // ...
  }
}
```

All other fields(like `contentType`, `cacheControl`, ...) are working fine.

But for `cacheExpiry`, `writeHttpMetadata()` throws `TypeError`.

> Method Date.prototype.toUTCString called on incompatible receiver function toUTCString() { [native code] }

- - -

Maybe related code?

https://github.com/cloudflare/workerd/blob/ee4a7b99526d53815dfb1c1c7b9417cdfa3b8aa6/src/workerd/api/r2-bucket.c%2B%2B#L45-L65

- - - 

This happens on both `wrangler dev` and `wrangler dev --remote`.

**Comment by jasnell** [comment]
Oh, that's strange. Definitely appears to be something wonky there... will get a ticket opened to take a look.

**Comment by ohodson** [comment]
Hi @leader22, thanks for raising this issue.

A fix for the problem (#868) can be found in the latest release:
* https://github.com/cloudflare/workerd/releases/tag/v1.20230717.0.

There's also an internal test that confirms the fix proposed in #868 addresses the issue.

Should you find any additional problems relating to this code path, please re-open this bug report.

Thank you.
